samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java

```diff
@@ -164,25 +170,69 @@ public SamzaSqlRelMessage convertToRelMessage(KV<Object, Object> samzaMessage) {
     return new SamzaSqlRelMessage(samzaMessage.getKey(), fieldNames, values);
   }
 
+  private SamzaSqlRelRecord convertToRelRecord(GenericData.Record avroRecord) {
+    List<Object> values = new ArrayList<>();
+    List<String> fieldNames = new ArrayList<>();
+    if (avroRecord != null) {
+      fieldNames.addAll(avroRecord.getSchema().getFields()
+          .stream()
+          .map(Schema.Field::name)
+          .collect(Collectors.toList()));
+      values.addAll(avroRecord.getSchema().getFields()
+          .stream()
+          .map(x -> {
```

Comment:
I don't think this will work if the sub record  has an Array. I think this conversion logic here should use AvroToRelObjConverter. Ideally this conversion shoudl recursively use the same conversion logic that we use for the top level.